velox/functions/sparksql/String.h

```diff
@@ -1181,4 +1181,56 @@ struct FindInSetFunction {
   }
 };
 
+template <typename T>
+struct SoundexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr bool is_default_ascii_behavior = true;
+  static constexpr int32_t reuse_strings_from_arg = 0;
```

Comment:
I assume it is rare when input is returned as is, hence, perhaps, it is not a good idea to carry original strings to the output.